
##glmapping parameter
#Awareness map radius d_Rho*n_Rho
#devition of orientation 360/d_Phi_deg
#n_Z = Z_below + Z_over + 1;
#Number of grid n_Rho*(360/d_Phi)*(nz_below+nz_over+1)

mlmapping_am_d_Rho:         0.20
mlmapping_am_d_Phi_deg:      5
mlmapping_am_d_Z:           0.20
mlmapping_am_n_Rho:          40
mlmapping_am_n_Z_below:      20
mlmapping_am_n_Z_over:       20

#Lolbalmap papameter
#n_xy/n_z available value     3x(2n+1)  3x(odd number)  E.g 27 39 45 75 81 135 165 195 243 495...
mlmapping_subbox_d_xyz:          0.2
mlmapping_subbox_n:              10   #number of cells on each axis of the subbox
mlmapping_lm_log_odds_min:      -2.0  # -2.0 for p = 0.01
mlmapping_lm_log_odds_max:       4.2  
mlmapping_lm_measurement_hit:    0.7  #0.7 for p = 0.83
mlmapping_lm_measurement_miss:  -0.9  # -0.4 for p = 0.28, -0.7 fpr p = 0.166, -1.0 for p = 0.09
mlmapping_lm_occupied_sh:        2.0  # 1.0 for p = 0.9, 2.0 for p = 0.99, 1.3 for p = 0.95

mlmapping_cam_cx: 212.6516265869
mlmapping_cam_cy: 117.238
mlmapping_cam_fx: 213.728866577
mlmapping_cam_fy: 213.728866577  # for real data

mlmapping_depth_noise_coe: 0.00375   # must > 0. noise model, the variance of depth at distance d is coe*d^2
# camera2odom_latency: 0.085 # the latency (second) of the depth image minus the latency of the odom (imu). It can be estimated by the method in Dr. Han Chen's PHD thesis
camera2odom_latency: 0.005 #

apply_odds_fusion_awareness: true
use_exploration_frontiers: false
visualize_odds: false
mlmapping_sample_cnt:       1000

mlmapping_inflate_n:        2 # inflate size: mlmapping_subbox_d_xyz * mlmapping_inflate_n. It must smaller than mlmapping_subbox_n
mlmapping_inflate_global_n: 3 # global inflate range: +-mlmapping_subbox_d_xyz * mlmapping_subbox_n * mlmapping_inflate_global_n
mlmapping_apply_inflate:    true
mlmapping_use_raycasting:   true
mlmapping_enable_esdfs:     false
mlmapping_enable_2d_grid:   false

publish_T_wb:      true
publish_T_bs:      true

##transformation from sensor frame to body frame
sensor_frame_id:   "camera_depth_optical_frame"
T_B_S:
[  0.0,   0.0,  1.0,  0.10,  #sensor position in body frame, x
  -1.0,   0.0,  0.0,  0.0,   #sensor position in body frame, y
   0.0,  -1.0,  0.0,  0.005,   #sensor position in body frame, z
   0.0,   0.0,  0.0,  1.0]


body_frame_id:      "body_frame"
awareness_frame_id: "awareness_frame"
local_frame_id:     "local_frame"
world_frame_id:     "map"

#generate projected 2D map
use_projected_2d_map:   true
use_relative_height:    true
projected_2d_map_min_z: -0.3
projected_2d_map_max_z:  0.3

#generate esdfs
use_esdfs: false
esdfs_n_xy:              35
esdfs_n_z:               13
batch_max_search_range:  10

#visulization
visulize_raycasting: false






